plugins/inputs/sysstat/sysstat.go

```diff
@@ -199,6 +205,17 @@ func (s *Sysstat) collect() error {
 	return nil
 }
 
+func createTempFile() (string, error) {
+	tmpPath := path.Join(os.TempDir(), "tg-sysstat")
+	os.Mkdir(tmpPath, os.ModeDir)
+	f, err := ioutil.TempFile(tmpPath, "sysstat")
+	if err != nil {
+		return "", fmt.Errorf("unable to create temp file in path %s: %w", tmpPath, err)
+	}
+	defer f.Close()
```

Comment:
If we close the file here, we won't be able to use it.  Best to run this on a real system.

Reply:
the goal is to let another process write into it so Telegraf can read it later

Reply:
There appears to be a race condition when Sophos antivirus tries to read the file and it is removed. Since the files are created and removed very quickly this is causing issues.  We would like to choose an arbitrary location for these files which we can then exclude from the active scans.

Reply:
Using a glob exclusion in Sophos such as /tmp/sysstat* does not work unfortunately, and we really would like to not have to exclude all of /tmp

Reply:
I'm bumping this up to the top as it's related to why this was opened.

Reply:
@takeahike I see, the old code is indeed not creating the file correctly because it uses the timestamp to make the file "random", this could be predicted and I'm not sure exactly what could be done, sometimes this is a security issue.  At any rate the current code is not follow best tempfile practices, we need to fix this but I wouldn't necessarily exclude it from the scan.

Reply:
@ssoroka Do we need a temporary file here at all?  Maybe sadf can write to stdout?

Reply:
that'd be a lot cleaner.

Reply:
It seems like it would be possible, read the stdout from sadc and feed it back in to all sadf processes on stdin,  but a bit larger change so we could do it later.

I also read in `man sadc`:
> In this case, if the file already exists, sadc will overwrite it if it is from a previous month.

So we don't really need to recreate the tempfile, we could just use the same one each iteration and then clean up the directory and the file once when the plugin stops.

Reply:
Sysstat's sadf program doesn't have the ability to read from stdin. It can read from an arbitrary file or daily files in a system location or in an arbitrary directory. On linux, if you try to force it to read from stdin by giving /proc/self/fd/0 as the filename, it fails with an "illegal seek" error.

It looks like this plugin is stuck with using temp files.